public/search.js

```diff
@@ -16,9 +16,11 @@ function getSelected(id) {
 
 $('#valDropdown')
   .change(function() {
-    $('#dragUp, #dragDown, .valTimesDisplay, #startValBox, #endValBox').css({'visibility': 'hidden'});
+    writeQuery();
+    $('#filledRect, #dragUp, #dragDown, .valTimesDisplay, #startValBox, #endValBox').css({'visibility': 'hidden'});
```

Comment:
rather than calling out all these ids, it would be better/simpler to specify a class ("time-controls", or something like that). Add the class to all of the items, then

    $('.time-controls').css({'visibility': 'hidden'});